builder/js/ncf.js

```diff
@@ -336,7 +336,92 @@ $scope.isSelected = function(technique) {
 $scope.isSelectedMethod = function(method) {
   return angular.equals($scope.selectedMethod,method);
 };
+$scope.getSessionStorage = function(){
+  $scope.resetFlags();
+  var matchIndex;
+  var t1,t2 = undefined;
+  t1 = JSON.parse(sessionStorage.getItem('selectedTechnique'));
+  t2 = JSON.parse(sessionStorage.getItem('originalTechnique'));
+  if(t2 !== null && t2.bundle_name === "") t2.bundle_name=undefined;
+  if(t1 !== null){
+    $scope.restoreFlag  = true;
+    $scope.selectedTechnique = angular.copy(t1);
+    $scope.originalTechnique = angular.copy(t2);
+    if($scope.originalTechnique.bundle_name !== undefined){
+    //Not a new technique
+      var deleted = true;
+      for(var i in $scope.techniques){
+        if(!$scope.conflictFlag && $scope.originalTechnique.bundle_name == $scope.techniques[i].bundle_name){
+          matchIndex = i;
+          //And the original technique still exists
+          if(t2.hasOwnProperty('saving'))$scope.techniques[i].saving  = false;
+          if(t2.hasOwnProperty('isClone'))$scope.techniques[i].isClone = false;
+          if(!angular.equals(t2, $scope.techniques[i])){
+            var modalInstance = $modal.open({
+              templateUrl: 'RestoreWarningModal.html',
+              controller: RestoreWarningModalCtrl,
+              backdrop : 'static',
+              resolve: {
+                technique: function () {
+                  return $scope.selectedTechnique;
+                }
+                , editForm  : function() { return  $scope.editForm }
+              }
+            });
+            modalInstance.result.then(function (doSave) {
+              if (doSave) {
+                $scope.originalTechnique = angular.copy($scope.techniques[matchIndex]);
+                $scope.selectedTechnique = angular.copy($scope.techniques[matchIndex]);
+                $scope.resetFlags();
+              }else{
+                $scope.tempOriginalTechnique = angular.copy($scope.techniques[matchIndex]);
```

Comment:
Why not $scope.originalTechnique ? 
What is the purpose of $scope.tempOriginalTechnique ?
You should always compare to stored state